core/src/main/java/org/apache/calcite/rel/rules/FilterJoinRule.java

```diff
@@ -116,6 +115,15 @@ protected FilterJoinRule(RelOptRuleOperand operand, String id,
 
   //~ Methods ----------------------------------------------------------------
 
+  /** Returns if it is needed to push the filter condition above join
+   * into the join condition.
+   */
+  private boolean needsPushInto(Join join) {
+    // If the join force the join info to be based on column equality,
+    // Or it is a non-correlated semijoin, returns false.
```

Comment:
A SemiJoin can't be correlated.

Reply:
@danny0405 Can you update the comment?

Reply:
@hsyuan Oops, i missed your comment, feel free to update it as what you want is to be.